src/controllers/places/add_place.js

```diff
@@ -25,23 +25,25 @@ module.exports = (req, res) => {
   getLatLng(req.body.address, (err, response) => {
     if (!err) {
       apiBody.location = [response.lat, response.lng];
-      console.log(apiBody);
     }
 
+    const url = req.body._method === 'put' ? `${placesURL}/${req.params.id}` : placesURL;
+
     const reqOptions = {
-      url: placesURL,
-      method: 'POST',
+      url: url,
+      method: req.body._method,
       body: apiBody,
       json: true
     };
-
-    Request(reqOptions, (error, postRes, postResBody) => {
+    Request(reqOptions, (error, apiResponse, apiResponseBody) => {
       if (error) {
-        console.log(error);
         res.send('huge failure');
+      }
+      if ((reqOptions.method === 'put' && apiResponse.statusCode !== 200) || (reqOptions.method === 'post' && apiResponse.statusCode !== 201)) {
+        res.status(404).send(apiResponseBody);
       } else {
-        console.log(postResBody);
-        res.redirect(`/${req.params.lang}/places`);
+        const urlEnd = req.body._method === 'put' ? `place/${req.params.id}` : 'places';
+        res.redirect(`/${req.params.lang}/${urlEnd}`);
       }
     });
   });
```

Comment:
Ok so, I know this is more like the way you originally had it so this is probably annoying, but on lines 30, 42 and 45 you ask the same question. instead, at the top of the file lets do:
```js
let url, urlEnd, correctResponseStatusCode
switch req.body._method
  case 'post'
    // set the above variables as they would be for post
  case 'put'
   // set the above variables as they would be for put
  default
   // this will happen when there is an error in your code so maybe send the 404 page as default
```

Then after this switch statement all the other stuff can be simplified